lib/rust/ensogl/core/src/display/garbage.rs

```diff
@@ -68,24 +68,27 @@ impl Collector {
     /// Pixel value requested (the points 1 and 2 in [`Collector`] docs).
     #[profile(Debug)]
     pub fn pixel_synced(&self) {
-        let mut garbage = self.garbage.borrow_mut();
-        let objects_being_moved = std::mem::take(&mut garbage.before_pixel_sync);
-        garbage.before_pixel_update.extend(objects_being_moved);
+        let garbage = &mut *self.garbage.borrow_mut();
+        garbage.before_pixel_update.append(&mut garbage.before_pixel_sync);
     }
 
     /// Pixel value retrieved (the point 3 in [`Collector`] docs).
     #[profile(Debug)]
     pub fn pixel_updated(&self) {
-        let mut garbage = self.garbage.borrow_mut();
-        let objects_being_moved = std::mem::take(&mut garbage.before_pixel_update);
-        garbage.before_mouse_events.extend(objects_being_moved);
+        let garbage = &mut *self.garbage.borrow_mut();
+        garbage.before_mouse_events.append(&mut garbage.before_pixel_update);
     }
```

Comment:
1. Why these changes? The previous code was a little bit more verbose but also more explanatory. I'm OK with the shorter version, although I like the earlier one a little bit more. Before making this change, talk with the code author aobut it, with Adam.
2. There is a logic change. The old code was removing `garbage.before_pixel_update`, the new code is not. Why? There is no explanation in this PR why such change was introduced and what happened here.

Reply:
[`Vec::append`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.append) also does remove the items from source vector. This is how it is a move, not a clone. That way it is not a logic change.

The actual significant change here is that the vector is not reallocated every time a garbage collector is run. Instead, the vector capacity remains allocated and is ready to be reused for future garbage.

Reply:
@farmaazon could you take a look at this part to verify that the changes are correct?

Reply:
I'm ok with this change. I just wasn't aware of `append` method.